### PR TITLE
Add formatting and bump sbt

### PR DESCRIPTION
**Pull request by mslabek**

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env sh
```
Comment (coutoPL) [inline .husky/pre-commit]: husky is a JS solution. We reused the existing hook from the fronted in Nu. Here, we can have sth like that:  `.hooks/pre-commit`

File: formatStagedBackendFiles.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env sh
```
Comment (coutoPL) [inline formatStagedBackendFiles.sh]: let's move the file to `scripts` folder

File: project/NuToolsGitPathsConfiguratorPlugin.scala [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+  override def globalSettings: Seq[Setting[_]] = Seq(
+    Global / onLoad                      := (Global / onLoad).value andThen configureNuToolsGitPathsOnLoad,
+    configureNuToolsGitPaths / aggregate := false
+  )
+
+  lazy val configureNuToolsGitPathsTask = Def.task {
+
+    implicit val logger: ManagedLogger = streams.value.log
+    updateGitConfigIfNeeded("core", "hooksPath", ".husky")
```
Comment (coutoPL) [inline project/NuToolsGitPathsConfiguratorPlugin.scala]: .husky

File: project/plugins.sbt [hunk 4]
```diff
@@ -1,2 +1,10 @@
-addSbtPlugin("com.eed3si9n" % "sbt-assembly" % "0.14.7")
-addSbtPlugin("net.virtual-void" % "sbt-dependency-graph" % "0.10.0-RC1")
+libraryDependencies ++= Seq(
+  "com.github.pathikrit" %% "better-files" % "3.9.2",
+  "com.lihaoyi"          %% "os-lib"       % "0.10.3",
```
Comment (coutoPL) [inline project/plugins.sbt:3]: is it used?

Comment (mslabek) [reply project/plugins.sbt:3]: yes, in `FormatStagedScalaFilesPlugin` on 43:5 in `getStagedScalaFiles` and on 65:5 `addToGitAllStagedFilesOnceAgain`. `better-files` is also used.

File: project/NuToolsGitPathsConfiguratorPlugin.scala (hunk 3 above)
Comment (mslabek) [reply project/NuToolsGitPathsConfiguratorPlugin.scala]: done :+1:

File: formatStagedBackendFiles.sh (hunk 2 above)
Comment (mslabek) [reply formatStagedBackendFiles.sh]: done :+1:

File: .husky/pre-commit (hunk 1 above)
Comment (mslabek) [reply .husky/pre-commit]: didnt know that, thanks. Done :+1:

**Review by mslabek — COMMENTED** [review]
Before merging I plan to squash the review fixes into the pre-format commit and amend the .git-blame-ignore-revs commit with the new hash.

**Review by coutoPL — APPROVED** [review]
LGTM

**Pull request merged**